### ISSUE TITLE
Error is thrown for assets with persistent cache enabled (The "path" argument must be of type string. Received undefined)

### ISSUE THREAD

**Issue by Mesoptier** [webdeveric/webpack-assets-manifest#182]
## The problem

Our builds sometimes fail on webpack-assets-manifest with the following error:
```
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined
at validateString (internal/validators.js:120:11)
at Object.dirname (path.js:1128:5)
at WebpackAssetsManifest.handleProcessAssetsAnalyse (/code/node_modules/webpack-assets-manifest/src/WebpackAssetsManifest.js:477:31)
at fn (/code/node_modules/webpack/lib/Compilation.js:425:10)
at _next3 (eval at create (/code/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:9:1)
at eval (eval at create (/code/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:25:1)
at runMicrotasks (<anonymous>)
at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

This error keeps occurring until I invalidate webpack's persistent cache. So it seems to me that something is going wrong in relation to said cache.

I believe there's been some similar issues to this, but those have been closed and the error is still happening, hence the new issue.

### Debugging steps

As is clear from the error message, this takes place in the `handleProcessAssetsAnalyse` method. I've added the following debugging code in that method:
```js
// START DEBUGGING
if (!filename) {
    console.log('chunk', chunk);
    console.log('module', module);
    console.log('info', info);

    const codeGenerationResults = compilation.codeGenerationResults;
    const codeGenResult = codeGenerationResults.get(
        module,
        chunk.runtime
    );
    console.log('codeGenResult: filename', codeGenResult.data.get("filename"));
    console.log('codeGenResult: assetInfo', codeGenResult.data.get("assetInfo"));
}
// END DEBUGGING
```

<details>
<summary>
Full log output (click to view)
</summary>

```
chunk Chunk {
  id: 'child',
  ids: [ 'child' ],
  debugId: 1028,
  name: 'child',
  idNameHints: SortableSet [Set] {
    _sortFn: undefined,
    _lastActiveSortFn: Symbol(not sorted),
    _cache: undefined,
    _cacheOrderIndependent: undefined
  },
  preventIntegration: false,
  filenameTemplate: undefined,
  _groups: SortableSet [Set] {
    Entrypoint {
      groupDebugId: 5028,
      options: [Object],
      _children: [SortableSet [Set]],
      _parents: [SortableSet [Set]],
      _asyncEntrypoints: [SortableSet [Set]],
      _blocks: [SortableSet [Set]],
      chunks: [Array],
      origins: [Array],
      _modulePreOrderIndices: [Map],
      _modulePostOrderIndices: [Map],
      index: 0,
      _runtimeChunk: [Circular],
      _entrypointChunk: [Circular],
      _initial: true
    },
    _sortFn: [Function: compareChunkGroupsByIndex],
    _lastActiveSortFn: Symbol(not sorted),
    _cache: undefined,
    _cacheOrderIndependent: undefined
  },
  runtime: 'child',
  files: SetDeprecatedArray [Set] {},
  auxiliaryFiles: Set {
    'fonts/37CB36_0_0.eot?bust=a9590edf12014650a66b',
    'fonts/37CB36_0_0.ttf?bust=799b01423a32a2057ae5',
    'fonts/37CB36_0_0.woff?bust=e2f7b2bf7d744847d59a',
    'fonts/37CB36_0_0.woff2?bust=5f0c3e4353c778c2327a',
    'fonts/37CB36_1_0.eot?bust=06dccbd21c156c3b2fc3',
    'fonts/37CB36_1_0.ttf?bust=7fa0fe6518a5975585ba',
    'fonts/37CB36_1_0.woff?bust=eca63048c4e5f75f5edf',
    'fonts/37CB36_1_0.woff2?bust=9f1671e1d819b9ad0d71',
    'fonts/37CB36_2_0.eot?bust=42c277ef94dfa870d45c',
    'fonts/37CB36_2_0.ttf?bust=04d0ac72d087e24f4d92',
    'fonts/37CB36_2_0.woff?bust=4d860a557c9add08755b',
    'fonts/37CB36_2_0.woff2?bust=c65cb371685ee4ba29c1',
    'img/badge-90.png?bust=ba0c2d488ccd554b5474',
    'fonts/font/fontello.eot?37800173?bust=c4013e415c4fab9fd7c6',
    'fonts/font/fontello.svg?37800173?bust=c462cb85609a6fe58f8f',
    'module-files/flag-icon-css/flags/4x3/at.svg?bust=7824152b6e660004d832',
    'module-files/flag-icon-css/flags/4x3/be.svg?bust=27d8ca49197f90010475',
    'module-files/flag-icon-css/flags/4x3/ch.svg?bust=9c26f60a63bf575c6b7b',
    'module-files/flag-icon-css/flags/4x3/de.svg?bust=3e726c2b6a59e6e4543c',
    'module-files/flag-icon-css/flags/4x3/dk.svg?bust=eb1416e02baeee91a39f',
    'module-files/flag-icon-css/flags/4x3/ee.svg?bust=6088c9ceb092913b54d7',
    'module-files/flag-icon-css/flags/4x3/es.svg?bust=cc1b41b0e9485796cebb',
    'module-files/flag-icon-css/flags/4x3/fi.svg?bust=2649533e1d44a2ef75d5',
    'module-files/flag-icon-css/flags/4x3/fr.svg?bust=b1156355de9691d768df',
    'module-files/flag-icon-css/flags/4x3/gb.svg?bust=4f1b7af5c0bae6aae85c',
    'module-files/flag-icon-css/flags/4x3/hu.svg?bust=0d7409f88bca8325938e',
    'module-files/flag-icon-css/flags/4x3/ie.svg?bust=c68ff961baf04c04f9be',
    'module-files/flag-icon-css/flags/4x3/it.svg?bust=8d15de04f5f6e8e89cab',
    'module-files/flag-icon-css/flags/4x3/nl.svg?bust=e336d50a0531bb958fa9',
    'module-files/flag-icon-css/flags/4x3/no.svg?bust=0b41df77e951a30bbfcc',
    'module-files/flag-icon-css/flags/4x3/pl.svg?bust=562edca5bb39d66f4c92',
    'module-files/flag-icon-css/flags/4x3/pt.svg?bust=c31a6c496e2c66def3f6',
    'module-files/flag-icon-css/flags/4x3/ru.svg?bust=0cacf46e6f473fa88781',
    'module-files/flag-icon-css/flags/4x3/se.svg?bust=fe725901338e5651e142',
    'module-files/flag-icon-css/flags/4x3/tr.svg?bust=ed6d5f37779af38911b0',
    'module-files/flag-icon-css/flags/4x3/us.svg?bust=ae65659236a7e3484027'
  },
  rendered: false,
  hash: '24924f30318c649582329f690675691c',
  contentHash: [Object: null prototype] { javascript: '18acf6a079edeeafe756' },
  renderedHash: '24924f30318c64958232',
  chunkReason: undefined,
  extraAsync: false
}
module NormalModule {
  dependencies: [],
  blocks: [],
  type: 'asset/resource',
  context: '/code/module/Landing/assets/fonts/font',
  layer: null,
  needId: true,
  debugId: 1348,
  resolveOptions: {},
  factoryMeta: undefined,
  useSourceMap: true,
  useSimpleSourceMap: false,
  _warnings: undefined,
  _errors: undefined,
  buildMeta: { exportsType: 'default' },
  buildInfo: {
    cacheable: true,
    parsed: true,
    fileDependencies: undefined,
    contextDependencies: undefined,
    missingDependencies: undefined,
    buildDependencies: undefined,
    valueDependencies: undefined,
    hash: 'efe31bfebe396cf4dec102785b369ea9',
    assets: undefined,
    assetsInfo: undefined,
    strict: true,
    dataUrl: false,
    snapshot: Snapshot {
      _flags: 3,
      startTime: 1633074157421,
      fileTimestamps: [Map],
      fileHashes: undefined,
      fileTshs: undefined,
      contextTimestamps: undefined,
      contextHashes: undefined,
      contextTshs: undefined,
      missingExistence: undefined,
      managedItemInfo: undefined,
      managedFiles: undefined,
      managedContexts: undefined,
      managedMissing: undefined,
      children: undefined
    }
  },
  presentationalDependencies: undefined,
  request: '/code/module/Landing/assets/fonts/font/fontello.eot?37800173',
  userRequest: '/code/module/Landing/assets/fonts/font/fontello.eot?37800173',
  rawRequest: '../font/fontello.eot?37800173',
  binary: true,
  parser: AssetParser { dataUrlCondition: false },
  parserOptions: undefined,
  generator: AssetGenerator {
    dataUrlOptions: undefined,
    filename: [Function: generateLegacyFilename],
    publicPath: undefined,
    emit: true
  },
  generatorOptions: { filename: [Function: generateLegacyFilename] },
  resource: '/code/module/Landing/assets/fonts/font/fontello.eot?37800173',
  resourceResolveData: undefined,
  matchResource: undefined,
  loaders: [],
  error: null,
  _source: RawSource {
    _valueIsBuffer: true,
    _value: <Buffer 30 3c 00 00 88 3b 00 00 01 00 02 00 00 00 00 00 02 00 05 03 00 00 00 00 00 00 01 00 90 01 00 00 00 00 4c 50 00 00 00 00 00 00 00 00 00 00 00 00 00 00 ... 15358 more bytes>,
    _valueAsBuffer: <Buffer 30 3c 00 00 88 3b 00 00 01 00 02 00 00 00 00 00 02 00 05 03 00 00 00 00 00 00 01 00 90 01 00 00 00 00 4c 50 00 00 00 00 00 00 00 00 00 00 00 00 00 00 ... 15358 more bytes>
  },
  _sourceSizes: undefined,
  _sourceTypes: Set { 'javascript', 'asset' },
  _lastSuccessfulBuildMeta: { exportsType: 'default' },
  _forceBuild: false,
  _isEvaluatingSideEffects: false,
  _addedSideEffectsBailout: undefined
}
info {
  sourceFilename: 'module/Landing/assets/fonts/font/fontello.eot?37800173'
}
codeGenResult: filename fonts/font/fontello.eot?37800173?bust=c4013e415c4fab9fd7c6
codeGenResult: assetInfo {
  sourceFilename: 'module/Landing/assets/fonts/font/fontello.eot?37800173',
  immutable: true,
  contenthash: 'c4013e415c4fab9fd7c6'
}
```
</details>

#### Looking for a cause for the error

First of all, I only log when the `filename` is `undefined` to prevent spamming the output, since that's when the error occurs.

I started with logging the values of the `chunk`, `module`, and `info` variables, to see what was actually going on. It turns out that the `filename` property is missing from `module.buildInfo` in this context:

```
module NormalModule {
  // ...
  buildInfo: {
    cacheable: true,
    parsed: true,
    fileDependencies: undefined,
    contextDependencies: undefined,
    missingDependencies: undefined,
    buildDependencies: undefined,
    valueDependencies: undefined,
    hash: 'efe31bfebe396cf4dec102785b369ea9',
    assets: undefined,
    assetsInfo: undefined,
    strict: true,
    dataUrl: false,
    snapshot: Snapshot {
      _flags: 3,
      startTime: 1633074157421,
      fileTimestamps: [Map],
      fileHashes: undefined,
      fileTshs: undefined,
      contextTimestamps: undefined,
      contextHashes: undefined,
      contextTshs: undefined,
      missingExistence: undefined,
      managedItemInfo: undefined,
      managedFiles: undefined,
      managedContexts: undefined,
      managedMissing: undefined,
      children: undefined
    }
  },
  // ...
}
```

So that explains why the error is thrown!

#### Looking for a solution

With that information I started scouring the webpack source code for the path by which `module.buildInfo.filename` would be set. I'm not sure if I necessarily found the answer to that question, but I did find this snippet of code in webpack's AssetModulesPlugin:

```js
filename:
    module.buildInfo.filename ||
    codeGenResult.data.get("filename"),
```
[Link to above snippet in webpack source](https://github.com/webpack/webpack/blob/d77d8636015ae71195b16b39dd0e6daebdcace83/lib/asset/AssetModulesPlugin.js#L181-L183)

Note that this looks like a similar context to the one webpack-assets-manifest is running in! So that's where the second part of my debugging code comes in: logging the values of `filename` and `assetInfo` as set in `compilation.codeGenerationResults`, just like in the snippet above.

Those logged values were:
```
codeGenResult: filename fonts/font/fontello.eot?37800173?bust=c4013e415c4fab9fd7c6
codeGenResult: assetInfo {
  sourceFilename: 'module/Landing/assets/fonts/font/fontello.eot?37800173',
  immutable: true,
  contenthash: 'c4013e415c4fab9fd7c6'
}
```

So it looks like the `filename` value from `compilation.codeGenerationResults` is usable, even in the context where `module.buildInfo.filename` would be `undefined`.

### A solution?
My suggestion for solving this is to add the same fallback to `codeGenerationResults` as in webpack's AssetModulesPlugin:

```diff
  /**
   * Record details of Asset Modules
   *
   * @param {*} compilation
   */
  handleProcessAssetsAnalyse( compilation /* , assets */ )
  {
    const { contextRelativeKeys } = this.options;
+   const codeGenerationResults = compilation.codeGenerationResults;

    for ( const chunk of compilation.chunks ) {
      const modules = compilation.chunkGraph.getChunkModulesIterableBySourceType(
        chunk,
        'asset',
      );

      if ( modules ) {
        for ( const module of modules ) {
-         const { assetInfo, filename } = module.buildInfo;
+         let { assetInfo, filename } = module.buildInfo;
+
+         const codeGenResult = codeGenerationResults.get(
+           module,
+           chunk.runtime
+         );
+         assetInfo = assetInfo || codeGenResult.data.get('assetInfo');
+         filename = filename || codeGenResult.data.get('filename');

          const info = Object.assign(
            {
              sourceFilename: path.relative( compilation.compiler.context, module.userRequest ),
            },
            assetInfo,
          );

          compilation.assetsInfo.set(filename, info);

          this.assetNames.set(
            contextRelativeKeys ?
              info.sourceFilename :
              path.join( path.dirname(filename), path.basename(module.userRequest) ),
            filename,
          );
        }
      }
    }
  }
```

### Webpack version

webpack: 5.45.1
webpack-assets-manifest: 5.0.6

### Webpack config

This is too large and complex to post here. Hopefully the debugging steps above, and link to webpack's AssetModulesPlugin source are enough to convince you that this is a real problem.

### Operating system

Ubuntu 20.04.3 LTS

**Comment by Mesoptier** [comment]
Oh, one more thing. We noticed that the filename of the specific asset in this case has two `?xxx`-parts. The first of those is set in the actual `import` in the LESS file where the font is included. The other is set with webpack's `generator.filename` option. It seems unlikely to me that this is what causes `module.buildInfo.filename` to not become set, but I figured I'd mention it to be safe.

**Comment by Mesoptier** [comment]
One more "one more thing": this comment in webpack's source code also seems to confirm that `module.buildInfo.XXX` doesn't get properly stored in combination with caching and instead "code generation results" should be used:

```
// Due to code generation caching module.buildInfo.XXX can't used to store such information
// It need to be stored in the code generation results instead, where it's cached too
// TODO webpack 6 For back-compat reasons we also store in on module.buildInfo
```
https://github.com/webpack/webpack/blob/beb42c64f696584ef570d8f57df448ddd0ac7238/lib/asset/AssetGenerator.js#L257-L258

**Comment by webdeveric** [comment]
@Mesoptier Thanks for the thorough debugging.

Does this only happen for you with asset modules? Or are other loaders also causing the issue?

**Comment by charle692** [comment]
Ok so this occurred to me a few times today, the first time I didn't have the logging changes that @Mesoptier wrote. I added those in so that the next time it occurs, I would have proper logs. 

The second time it occurred, I could see that the entrypoint processed in my case was a `*.js` file. In this case, it also seems like the issues occurred while processing an asset module, in this case a GIF.

I'll keep the logs in and share future occurrences. If there's anything specific you'd like me to log, I can modify things and report back with logs during the next occurrence. 

<details>
  <summary>Logs, click to view</summary>
 
  ```
  chunk Chunk {
  id: ‘[redacted]’,
  ids: [
    ‘[redacted]’
  ],
  debugId: 1520,
  name: undefined,
  idNameHints: SortableSet [Set] {
    _sortFn: undefined,
    _lastActiveSortFn: Symbol(not sorted),
    _cache: undefined,
    _cacheOrderIndependent: undefined
  },
  preventIntegration: false,
  filenameTemplate: undefined,
  _groups: SortableSet [Set] {
    ChunkGroup {
      groupDebugId: 5289,
      options: {},
      _children: [SortableSet [Set]],
      _parents: [SortableSet [Set]],
      _asyncEntrypoints: [SortableSet [Set]],
      _blocks: [SortableSet [Set]],
      chunks: [Array],
      origins: [Array],
      _modulePreOrderIndices: [Map],
      _modulePostOrderIndices: [Map],
      index: 88
    },
    _sortFn: [Function: compareChunkGroupsByIndex],
    _lastActiveSortFn: Symbol(not sorted),
    _cache: undefined,
    _cacheOrderIndependent: undefined
  },
  runtime: 'runtime',
  files: SetDeprecatedArray [Set] {
    'js/[redacted]-496b54.js'
  },
  auxiliaryFiles: Set { 'media/images/follow-up.gif' },
  rendered: false,
  hash: '1a95eac69f8f5c4cdba49602bf8fcbef',
  contentHash: [Object: null prototype] { javascript: 'e74318001d29691f384b' },
  renderedHash: '1a95eac69f8f5c4cdba4',
  chunkReason: undefined,
  extraAsync: false
}
module NormalModule {
  dependencies: [],
  blocks: [],
  parent: undefined,
  type: 'asset/resource',
  context: ‘[redacted]’,
  layer: null,
  needId: true,
  debugId: 26792,
  resolveOptions: {},
  factoryMeta: { sideEffectFree: true },
  useSourceMap: true,
  useSimpleSourceMap: false,
  _warnings: undefined,
  _errors: undefined,
  buildMeta: { exportsType: 'default' },
  buildInfo: {
    cacheable: true,
    parsed: true,
    fileDependencies: undefined,
    contextDependencies: undefined,
    missingDependencies: undefined,
    buildDependencies: undefined,
    valueDependencies: undefined,
    hash: 'fb677cbe291dfa14add56fa3ef4334bf',
    assets: undefined,
    assetsInfo: undefined,
    strict: true,
    dataUrl: false,
    snapshot: Snapshot {
      _flags: 3,
      startTime: 1634830231247,
      fileTimestamps: [Map],
      fileHashes: undefined,
      fileTshs: undefined,
      contextTimestamps: undefined,
      contextHashes: undefined,
      contextTshs: undefined,
      missingExistence: undefined,
      managedItemInfo: undefined,
      managedFiles: undefined,
      managedContexts: undefined,
      managedMissing: undefined,
      children: undefined
    }
  },
  presentationalDependencies: undefined,
  request: ‘[redacted]/follow-up.gif',
  userRequest: ‘[redacted]/follow-up.gif',
  rawRequest: './follow-up.gif',
  binary: true,
  parser: AssetParser { dataUrlCondition: false },
  parserOptions: undefined,
  generator: AssetGenerator {
    dataUrlOptions: undefined,
    filename: 'media/images/[name][ext][query]',
    publicPath: undefined,
    emit: true
  },
  generatorOptions: { filename: 'media/images/[name][ext][query]' },
  resource: ‘[redacted]/follow-up.gif',
  resourceResolveData: undefined,
  matchResource: undefined,
  loaders: [],
  error: null,
  _source: RawSource {
    _valueIsBuffer: true,
    _value: <Buffer 47 49 46 38 37 61 94 01 44 01 e6 00 00 00 00 00 04 05 08 08 07 0b 07 09 0b 09 0a 0f 0b 0c 12 0e 11 17 12 15 1a 17 18 1e 18 1a 1d 1a 1c 21 1e 22 23 21 ... 646056 more bytes>,
    _valueAsBuffer: <Buffer 47 49 46 38 37 61 94 01 44 01 e6 00 00 00 00 00 04 05 08 08 07 0b 07 09 0b 09 0a 0f 0b 0c 12 0e 11 17 12 15 1a 17 18 1e 18 1a 1d 1a 1c 21 1e 22 23 21 ... 646056 more bytes>
  },
  _sourceSizes: Map { 'javascript' => 42, 'asset' => 646106, undefined => 42 },
  _sourceTypes: Set { 'javascript', 'asset' },
  _lastSuccessfulBuildMeta: { exportsType: 'default' },
  _forceBuild: false,
  _isEvaluatingSideEffects: false,
  _addedSideEffectsBailout: undefined
}
info {
  sourceFilename: ‘[redacted]/follow-up.gif'
}
codeGenResult: filename media/images/follow-up.gif
codeGenResult: assetInfo {
  sourceFilename: ‘[redacted]/follow-up.gif'
}
<e> [webpack-dev-middleware] HookWebpackError: The "path" argument must be of type string. Received type undefined
<e>     at makeWebpackError ([redacted]/node_modules/webpack/lib/HookWebpackError.js:48:9)
<e>     at [redacted]/node_modules/webpack/lib/Compilation.js:2974:12
<e>     at eval (eval at create ([redacted]/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:17:1)
<e>     at fn ([redacted]/node_modules/webpack/lib/Compilation.js:476:17)
<e>     at _next2 (eval at create ([redacted]/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:14:1)
<e>     at eval (eval at create ([redacted]/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:34:1)
<e>     at runMicrotasks (<anonymous>)
<e>     at processTicksAndRejections (internal/process/task_queues.js:93:5)
<e> -- inner error --
<e> TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undefined
<e>     at validateString (internal/validators.js:112:11)
<e>     at Object.dirname (path.js:1129:5)
<e>     at WebpackAssetsManifest.handleProcessAssetsAnalyse ([redacted]/node_modules/webpack-assets-manifest/src/WebpackAssetsManifest.js:491:31)
<e>     at fn ([redacted]/node_modules/webpack/lib/Compilation.js:474:10)
<e>     at _next2 (eval at create ([redacted]/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:14:1)
<e>     at eval (eval at create ([redacted]/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:34:1)
<e>     at runMicrotasks (<anonymous>)
<e>     at processTicksAndRejections (internal/process/task_queues.js:93:5)
<e> caused by plugins in Compilation.hooks.processAssets
<e> TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undefined
<e>     at validateString (internal/validators.js:112:11)
<e>     at Object.dirname (path.js:1129:5)
<e>     at WebpackAssetsManifest.handleProcessAssetsAnalyse ([redacted]/node_modules/webpack-assets-manifest/src/WebpackAssetsManifest.js:491:31)
<e>     at fn ([redacted]/node_modules/webpack/lib/Compilation.js:474:10)
<e>     at _next2 (eval at create ([redacted]/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:14:1)
<e>     at eval (eval at create ([redacted]/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:34:1)
<e>     at runMicrotasks (<anonymous>)
<e>     at processTicksAndRejections (internal/process/task_queues.js:93:5)
  ```
</details>

**Comment by jpnauta-tugboat** [comment]
My team consistently runs into this issue :frowning_face: reproducing the issue consistently is difficult, but in general the steps to reproduce are:

- Start your local dev server using `webpack-assets-manifest`
- Enable [webpack cache](https://webpack.js.org/configuration/cache/) with type `filesystem` and a `idleTimeout` of a few minutes
- Change Git branches a few times while you wait for the cache to invalidate
- Restart your local dev start after cache is invalidated

Here's the stack trace I've seen, in case it helps :smile: 

![image](https://user-images.githubusercontent.com/77482660/139739113-d6de63a7-3bd1-4fdd-84a0-a3d8e6d4fab9.png)

**Comment by webdeveric** [comment]
I've made some updates to my `dev` branch. Can you install from there and see if it works for you?

```shell
npm install webdeveric/webpack-assets-manifest#dev
```

**Comment by Mesoptier** [comment]
@webdeveric Thanks for the update. We haven't had any crashes for about a month now, so it's going to be hard for me to tell if your changes fixed it. It certainly looks like it should fix it, though! I'm installing it now, and then we'll see if any more crashes occur.

---

@charle692 You mentioned you could semi-regularly replicate this issue. Is that still the case? If so, could you also try the branch mentioned above, to see if that fixes it?

**Comment by charle692** [comment]
Taking a look now. I can't replicate 100% but I will at least pull down the latest version and work with it for a while. If I don't experience the issue within a reasonable amount of time I would say it should be good. 

I'll report back regardless :).

**Comment by charle692** [comment]
Ok, so I was able to reproduce the issue almost immediately with the current latest release. I then cleared the cache and pull down your dev branch. Performed the same steps I did with the current release and the error no longer occurs. 

I continued to switch into all kinds of different branches with the webpack instance still running and I can no longer produce the error with your changes. 

So all that to say that your changes seem to have fixed the issue :) thanks @webdeveric!
